sentence_transformers/cross_encoder/losses/ListMLELoss.py

```diff
@@ -0,0 +1,290 @@
+        super().__init__()
+        self.model = model
+        self.lambda_weight = lambda_weight
+        self.activation_fct = activation_fct or nn.Identity()
+        self.mini_batch_size = mini_batch_size
+        self.respect_input_order = respect_input_order
+        self.eps = 1e-10
+
+        if self.model.num_labels != 1:
+            raise ValueError(
+                f"{self.__class__.__name__} supports a model with 1 output label, "
+                f"but got a model with {self.model.num_labels} output labels."
+            )
+
+    def forward(self, inputs: list[list[str], list[list[str]]], labels: list[Tensor]) -> Tensor:
+        """
+        Compute ListMLE loss for a batch of queries and their documents.
+
+        Args:
+            inputs: List of (queries, documents_list)
+            labels: Ground truth relevance scores, shape (batch_size, num_documents)
+
+        Returns:
+            Tensor: Mean ListMLE loss over the batch
+        """
+        if isinstance(labels, Tensor):
+            raise ValueError(
+                "ListMLELoss expects a list of labels for each sample, but got a single value for each sample."
+            )
+
+        if len(inputs) != 2:
+            raise ValueError(
+                f"ListMLELoss expects two inputs (queries, documents_list), but got {len(inputs)} inputs."
+            )
+
+        queries, docs_list = inputs
+        docs_per_query = [len(docs) for docs in docs_list]
+        max_docs = max(docs_per_query)
+        batch_size = len(queries)
+
+        if docs_per_query != [len(labels) for labels in labels]:
+            raise ValueError(
+                f"Number of documents per query in inputs ({docs_per_query}) does not match number of labels per query ({[len(labels) for labels in labels]})."
+            )
+
+        pairs = [(query, document) for query, docs in zip(queries, docs_list) for document in docs]
+
+        if not pairs:
+            # Handle edge case where there are no documents
+            return torch.tensor(0.0, device=self.model.device, requires_grad=True)
+
+        mini_batch_size = self.mini_batch_size or batch_size
+        if mini_batch_size <= 0:
+            mini_batch_size = len(pairs)
+
+        logits_list = []
+        for i in range(0, len(pairs), mini_batch_size):
+            mini_batch_pairs = pairs[i : i + mini_batch_size]
+
+            tokens = self.model.tokenizer(
+                mini_batch_pairs,
+                padding=True,
+                truncation=True,
+                return_tensors="pt",
+            )
+            tokens = tokens.to(self.model.device)
+
+            logits = self.model(**tokens)[0].view(-1)
+            logits_list.append(logits)
+
+        logits = torch.cat(logits_list, dim=0)
+        logits = self.activation_fct(logits)
+
+        # Create output tensor filled with a very small value for padded logits
+        logits_matrix = torch.full((batch_size, max_docs), -1e16, device=self.model.device)
```

Comment:
This is not a very small value, it's a very large negative value (-10000000000000000). cc @milistu this is also the case for e.g. ListNetLoss, although it's ignored via the labels.

Reply:
You're right. This should be `1e-16`. Would be same for ListNetLoss since I copied a lot of values from it.

Reply:
Wouldn't using a large negative value (-1e16) be safer for padded positions rather than a small positive value (1e-16)? The large negative value ensures padded positions have effectively zero contribution in softmax operations, while a small positive value might still have some minor influence.